core/src/com/po/fuck/GeometryMisc.java

```diff
@@ -39,22 +38,16 @@ public static <A extends Entity, B extends Entity> Vector2 direction(A from, B t
     /**
      * Finds the closest entity to a given main entity from a list of other
      * entities.
-     *
-     * @param <A>  The type of the main entity, extending class C.
-     * @param <B>  The type of the other entities in the list, extending class C.
-     * @param <C>  The common base class for main entity and other entities,
-     *             extending class Entity.
      * @param main The main entity.
      * @param list The list of other entities.
      * @return The closest entity to the main entity from the list, or null if the
      *         list is empty.
      */
-    public static <A extends C, B extends C, C extends Entity> C closest(A main, ArrayList<B> list) {
-        C closestEntity = null;
-        for (B entity : list) {
-            if (closestEntity == null || distance(main, closestEntity) > distance(main, entity)) {
+    public static<A extends Entity> A closest(A main, List<A> list) {
+        A closestEntity = null;
+        for (A entity : list) {
+            if (closestEntity == null || distance(main, closestEntity) > distance(main, entity))
                 closestEntity = entity;
-            }
         }
         return closestEntity;
     }
```

Comment:
Again, it should be like `public static<A extends Entity> A closest(A main, List<? extends Entity> list)`.

Reply:
Then it should not return `A`

Reply:
@RyhorKoshman fix this please. It should return Entity then.